stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -523,45 +532,50 @@ function complete_methods(ex_org::Expr, context_module::Module=Main)
     funct, found = get_type(ex_org.args[1], context_module)::Tuple{Any,Bool}
     !found && return out
 
-    args_ex, kwargs_ex = complete_methods_args(ex_org.args[2:end], ex_org, context_module, true, true)
-    push!(args_ex, Vararg{Any})
-    complete_methods!(out, funct, args_ex, kwargs_ex, MAX_METHOD_COMPLETIONS)
+    args_ex, kwargs_ex, kwargs_flag = complete_methods_args(ex_org.args[2:end], ex_org, context_module, true, true)
+    kwargs_flag == 2 && return out # one of the kwargs is invalid
+    kwargs_flag == 0 && push!(args_ex, Vararg{Any}) # allow more arguments if there is no semicolon
+    complete_methods!(out, funct, args_ex, kwargs_ex, MAX_METHOD_COMPLETIONS, kwargs_flag == 1)
 
     return out
 end
 
 MAX_ANY_METHOD_COMPLETIONS::Int = 10
 function complete_any_methods(ex_org::Expr, callee_module::Module, context_module::Module, moreargs::Bool, shift::Bool)
     out = Completion[]
-    args_ex, kwargs_ex = try
+    args_ex, kwargs_ex, kwargs_flag = try
         # this may throw, since we set default_any to false
         complete_methods_args(ex_org.args[2:end], ex_org, context_module, false, false)
     catch ex
         ex isa ArgumentError || rethrow()
         return out
     end
+    kwargs_flag == 2 && return out # one of the kwargs is invalid
+
+    # moreargs determines whether to accept more args, independently of the presence of a
+    # semicolon for the ".?(" syntax
     moreargs && push!(args_ex, Vararg{Any})
 
     seen = Base.IdSet()
     for name in names(callee_module; all=true)
-        if !Base.isdeprecated(callee_module, name) && isdefined(callee_module, name)
+        if !Base.isdeprecated(callee_module, name) && isdefined(callee_module, name) && !startswith(string(name), "#kw")
```

Comment:
Oh you're absolutely right, I wanted to remove internal functions like kwsorters and such, but in the end this does not make sense as you point out. It can either be replaced by `!startswith(string(name), '#')`, or we can drop that condition altogether and let internal functions appear. I don't think internal functions should appear here (if you want to access them, REPL completion is probably not the best entry point) but I don't mind dropping the condition if you think otherwise?